File: 프로그래머스/1/133499. 옹알이 （2）/옹알이 （2）.py
def solution(babbling):
    count = 0
    
    can_speak = ["aya", "ye", "woo", "ma"]
    
    for word in babbling:
        is_valid = True
        for sound in can_speak:
            if (sound * 2) in word:
                is_valid = False
                break
            word = word.replace(sound, " ")
        
        if is_valid and word.strip() == "":
            count += 1
    
    return count

File: 프로그래머스/1/133499. 옹알이 （2）/옹알이 （2）.js
const CAN_SPEAK = ["aya", "ye", "woo", "ma"];

function solution(babbling) {
    let count = 0;
    
    for (let word of babbling) {
        let isValid = true;
        
        for (let sound of CAN_SPEAK) {
            if (word.includes(sound.repeat(2))) {
                isValid = false;
                break;
            }
            word = word.replaceAll(sound, " ");
        }
        
        if (isValid && word.trim() === "") {
            count++;
        }
    }
    
    return count;
}